CHANGELOG.md

```diff
@@ -1,36 +1,51 @@
+## 0.14.2
+### Sept 9, 2025
+
+BUG FIXES:
+* Refresh the connection upon expiration (https://github.com/hashicorp/vault-plugin-database-snowflake/pull/134)
+
```

Comment:
We should follow the new changelog process: https://github.com/hashicorp/vault-plugin-database-snowflake/blob/main/README.md#updating-the-changelog